earth_enterprise/src/scons/getversion.py

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env python
+import os
+import argparse
+import git
+from datetime import datetime
+
+def GetVersion(backupFile, label=''):
+    """As getLongVersion(), but only return the leading *.*.* value."""
+
+    raw = GetLongVersion(backupFile, label)
+    final = raw.split("-")[0]
+
+    return final
+
+
+def GetLongVersion(backupFile, label=''):
+    """Create a detailed version string based on the state of
+    the software, as it exists in the repository."""
+ 
+    if _CheckGitAvailable():
+        ret = _GitGeneratedLongVersion()
+
+  # Without git, must use the backup file to create a string.
+    else:
+        base = _ReadBackupVersionFile(backupFile)
+        ret = '-'.join([base, _GetDateString()])
+
+  # Append the label, if there is one.
+    if len(label):
+        ret = '.'.join([ret, label])
+
+    return ret
+
+
+def _GitGeneratedLongVersion():
+    """Take the raw information parsed by git, and use it to
+       generate an appropriate version string for GEE."""
+
+
+    # For tagged commits, return the tag itself
+    if _IsCurrentCommitTagged():
+        return _VersionForTaggedHead()
+    else:
+        return _VersionFromTagHistory()
+
+
+def _IsCurrentCommitTagged():
+    """True if the current commit is tagged, otherwise False"""
+    raw = _GetCommitRawDescription()
+    # If this condition hits, then we are currently on a tagged commit.
+    return (len(raw.split("-")) < 4)
```

Comment:
An alternative way of doing this which would allow a little more flexibility in tag naming: `git describe --exact-match HEAD` will return the tag name if the current commit is tagged and will fail otherwise.  There's nothing wrong with what is here and other places also have the tag scheme built in.

Reply:
Thanks -- will leave code as-is, but that's a great tip on git describe.